scripts/stage1.sh

```diff
@@ -65,11 +65,12 @@ PACKAGES="dropbear-bin \
 	kmod \
 	systemd \
 	udev \
-	libssl1.1"
+	libssl1.1 \
+	lighttpd"
 
 apt-get download $(apt-cache depends --recurse --no-recommends --no-suggests \
 	--no-conflicts --no-breaks --no-replaces --no-enhances \
-	${PACKAGES} | grep "^\w")
+	${PACKAGES} | grep "^\w" | grep -v libfam0)
```

Comment:
Why are we excluding `libfam0`?

Reply:
*sigh* This is a tale of woe and misery. Until very recently, there was a bug in ubuntu where the lighttpd package didn't depend on `gamin`, which is in fact required for it to run correctly. This was corrected in the last week or so (in 18.04), and now it requires `libfam0` or `gamin`.  These two packages provide the same shared library, with slightly different capabilities. When you use apt-get install, the system just grabs the first from the list of alternatives, which is `gamin`. The apt-cache depends approach that we use here to effectively reimplement apt-get install, however, recurses through both options, rather than just selecting one.  If we fetch `libfam` as well as `gamin`, `libfam` puts its version in a location that ldd will find before the `gamin` version, which is the one we actually need.  As a result, we need to filter out the `libfam` package and its contents somewhere in the process if we want `lighttpd` to work correctly in the recovery environment.

Reply:
Interesting! I'm actually quite familiar with this particular misery as we had to go through similar pains with Prakash when figuring out how to get a list of dependencies for the Delphix upgrade image.

In any case, the info you posted here is very useful, please include it as a comment in the code.